@types/index.d.ts

```diff
@@ -0,0 +1,116 @@
+    getItemByName(name: string | number | string[] | number[]): Promise<object>;
+    getItemAttributeByName(name: string | number | string[] | number[]): Promise<object>;
+    getItemCategoryByName(name: string | number | string[] | number[]): Promise<object>;
+    getItemFlingEffectByName(name: string | number | string[] | number[]): Promise<object>;
+    getItemPocketByName(name: string | number | string[] | number[]): Promise<object>;
+    getMachineById(id: number | number[]): Promise<object>;
+    getMoveByName(name: string | number | string[] | number[]): Promise<object>;
+    getMoveAilmentByName(name: string | number | string[] | number[]): Promise<object>;
+    getMoveBattleStyleByName(name: string | number | string[] | number[]): Promise<object>;
+    getMoveCategoryByName(name: string | number | string[] | number[]): Promise<object>;
+    getMoveDamageClassByName(name: string | number | string[] | number[]): Promise<object>;
+    getMoveLearnMethodByName(name: string | number | string[] | number[]): Promise<object>;
+    getMoveTargetByName(name: string | number | string[] | number[]): Promise<object>;
+    getLocationByName(name: string | number | string[] | number[]): Promise<object>;
+    getLocationAreaByName(name: string | number | string[] | number[]): Promise<object>;
+    getPalParkAreaByName(name: string | number | string[] | number[]): Promise<object>;
+    getRegionByName(name: string | number | string[] | number[]): Promise<object>;
+    getAbilityByName(name: string | number | string[] | number[]): Promise<object>;
+    getCharacteristicById(id: number | number[]): Promise<object>;
+    getEggGroupByName(name: string | number | string[] | number[]): Promise<object>;
+    getGenderByName(name: string | number | string[] | number[]): Promise<object>;
+    getGrowthRateByName(name: string | number | string[] | number[]): Promise<object>;
+    getNatureByName(name: string | number | string[] | number[]): Promise<object>;
+    getPokeathlonStatByName(name: string | number | string[] | number[]): Promise<object>;
+    getPokemonByName(name: string | number | string[] | number[]): Promise<object>;
+    getPokemonEncounterAreasByName(name: string | number | string[] | number[]): Promise<object>;
+    getPokemonColorByName(name: string | number | string[] | number[]): Promise<object>;
+    getPokemonFormByName(name: string | number | string[] | number[]): Promise<object>;
+    getPokemonHabitatByName(name: string | number | string[] | number[]): Promise<object>;
+    getPokemonShapeByName(name: string | number | string[] | number[]): Promise<object>;
+    getPokemonSpeciesByName(name: string | number | string[] | number[]): Promise<object>;
+    getStatByName(name: string | number | string[] | number[]): Promise<object>;
+    getTypeByName(name: string | number | string[] | number[]): Promise<object>;
+    getLanguageByName(name: string | number | string[] | number[]): Promise<object>;
+
+    getEndpointsList(): Promise<object>;
```

Comment:
All these methods support passing a `dict` optional parameter:

https://github.com/PokeAPI/pokeapi-js-wrapper#root-endpoints-list

```json
{
  offset: 34,
  limit: 10,
}
```

Is it possible to document it?

Reply:
Sure, but how about the `resource` method? Does it also have the `dict` parameter?

Reply:
Alright, It's all done.